### PR TITLE
Fixing lint issues in the signature package

### PR DESCRIPTION
**Pull request by susperius**
#### Summary
Fixing pylint and pytype issues in the signature package

#### Release Note

None

#### Documentation

None

### CODE REVIEW

File: model_signing/signature/encoding.py
Comment (mihaimaruseac) [inline model_signing/signature/encoding.py:21]: Can we disable this locally and re-enable as soon as we no longer need it disabled?

File: model_signing/signature/key.py
Comment (mihaimaruseac) [inline model_signing/signature/key.py:99]:
```suggestion
            statement_pb.Statement()  # pylint: disable=no-member
        )
```

File: model_signing/signature/key.py
Comment (mihaimaruseac) [inline model_signing/signature/key.py:107]:
```suggestion
                'signature verification failed ' + str(e)
            ) from e
```

File: model_signing/signature/pki.py
Comment (mihaimaruseac) [inline model_signing/signature/pki.py:157]:
```suggestion
                f'signing certificate verification failed: {err}'
            ) from err
```

File: model_signing/signature/sigstore.py
Comment (mihaimaruseac) [inline model_signing/signature/sigstore.py:41]:
```suggestion
            id_provider: str | None = None
        ):
```

File: model_signing/signature/sigstore.py
Comment (mihaimaruseac) [inline model_signing/signature/sigstore.py:105]:
```suggestion
            raise VerificationError(str(e)
        ) from e
```

File: model_signing/signature/key.py
Comment (mihaimaruseac) [reply model_signing/signature/key.py:99]: This is to try and be consistent with the python style, but given that the linter didn't pick it up and we don't yet have an autoformatter, it's mostly a nit.

**Review by mihaimaruseac — COMMENTED** [review]
Thank you! This LGTM.

Added a few nits to simplify work later when we add more CI support.

File: model_signing/signature/encoding.py
Comment (mihaimaruseac) [inline model_signing/signature/encoding.py:21]:
```suggestion
def pae(
    statement: statement_pb.Statement  # pylint: disable=no-member
) -> bytes:
```

File: model_signing/signature/sigstore.py
Comment (mihaimaruseac) [inline model_signing/signature/sigstore.py:107]:
```suggestion
            raise VerificationError(str(e)) from e
```

**Review by mihaimaruseac — COMMENTED** [review]
Thank you!

**Comment by mihaimaruseac** [comment]
Oh, it seems trying to speed up merging by committing code suggestions actually makes it so that my review doesn't count for merging. Let's wait for @spencerschrock or @laurentsimon for one more review for merging this.

Sorry for the delay.

**Review by spencerschrock — APPROVED** [review]

**Pull request merged**
